vault/core.go

```diff
@@ -1317,10 +1317,10 @@ func NewCore(conf *CoreConfig) (*Core, error) {
 
 	c.limiterRegistry = conf.LimiterRegistry
 	c.limiterRegistryLock.Lock()
-	if conf.DisableRequestLimiter {
-		c.limiterRegistry.Disable()
-	} else {
+	if conf.EnableRequestLimiter {
 		c.limiterRegistry.Enable()
+	} else {
+		c.limiterRegistry.Disable()
```

Comment:
Nit: could simplify to this

```suggestion
        c.limiterRegistry.Disable()
	if conf.EnableRequestLimiter {
		c.limiterRegistry.Enable()
	}
```

Reply:
I think that might not be quite right, since it could nil out the old limiter automatically on reload (if it was previously enabled), discarding old measurements.